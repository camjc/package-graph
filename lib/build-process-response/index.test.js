const buildProcessResponse = require(".");

it("maps to graphviz dot format", () => {
  const body = JSON.stringify({
    data: {
      organization: {
        repositories: {
          nodes: [
            {
              content: {
                text: undefined
              },
              isArchived: false,
              isPrivate: true
            },
            {
              content: {
                text: JSON.stringify({
                  name: "ahm-partner-sales",
                  dependencies: {
                    "@ahmdigital/ui": "1.0.0",
                    lodash: "4.0.0"
                  }
                })
              },
              isArchived: true,
              isPrivate: true
            },
            {
              content: {
                text: JSON.stringify({
                  name: "ahm-sales",
                  dependencies: {
                    "@ahmdigital/ui": "1.0.0",
                    lodash: "4.0.0"
                  }
                })
              },
              isArchived: false,
              isPrivate: true,
              pullRequests: {
                nodes: [
                  {
                    author: {
                      login: "camjc"
                    }
                  },
                  {
                    author: {
                      login: "renovate"
                    }
                  },
                  {
                    author: {
                      login: "renovate"
                    }
                  }
                ]
              },
              url: "https://github.com/ahmdigital/ahm-sales"
            },
            {
              content: {
                text: JSON.stringify({
                  name: "ahm-members",
                  dependencies: {
                    "@ahmdigital/ui": "1.0.0",
                    "@ahmdigital/constants": "2.0.0"
                  },
                  devDependencies: { "@ahmdigital/eslint-config": "4.0.0" }
                })
              },
              isArchived: false,
              isPrivate: false,
              url: "https://github.com/ahmdigital/members"
            },
            {
              content: {
                text: JSON.stringify({
                  name: "@ahmdigital/ui",
                  version: "1.5.0",
                  dependencies: {
                    "@ahmdigital/constants": "2.0.0"
                  },
                  devDependencies: { "@ahmdigital/eslint-config": "4.0.0" }
                })
              },
              isArchived: false,
              isPrivate: false,
              url: "https://github.com/ahmdigital/ui"
            }
          ]
        }
      }
    }
  });
  const processResponse = buildProcessResponse("ahmdigital");
  expect(processResponse(body)).toMatchSnapshot();
});
