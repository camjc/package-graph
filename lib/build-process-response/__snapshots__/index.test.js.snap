// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maps to graphviz dot format 1`] = `
"
digraph {
  overlap=scalexy;
  splines=true;
  layout=neato;
  graph [ fontname=\\"gill sans\\" ];
  node [ fontname=\\"gill sans\\" ];
  edge [ fontname=\\"gill sans\\" ];

\\"@ahmdigital/ui\\" -> \\"ahm-sales\\" [ color=\\"#880000\\" ];
\\"@ahmdigital/ui\\" -> \\"ahm-members\\" [ color=\\"#880000\\" ];
\\"@ahmdigital/constants\\" -> \\"ahm-members\\"
\\"@ahmdigital/eslint-config\\" -> \\"ahm-members\\" [ style=\\"dashed\\" ];
\\"@ahmdigital/constants\\" -> \\"@ahmdigital/ui\\"
\\"@ahmdigital/eslint-config\\" -> \\"@ahmdigital/ui\\" [ style=\\"dashed\\" ];
\\"ahm-sales\\" [ href=\\"https://github.com/ahmdigital/ahm-sales/pulls\\",color=\\"#4b0000\\",label=\\"ahm-sales
renovate: 2
camjc: 1\\",shape=\\"box\\" ];
\\"ahm-members\\" [ href=\\"https://github.com/ahmdigital/members/pulls\\",color=\\"#00000\\",label=\\"ahm-members
\\" ];
\\"@ahmdigital/ui\\" [ href=\\"https://github.com/ahmdigital/ui/pulls\\",color=\\"#00000\\",label=\\"@ahmdigital/ui
1.5.0
\\" ];
}"
`;
